File: examples/python/delete_memory.py
import json
import sys
from urllib.parse import quote

from memside_http import api_request


if len(sys.argv) != 3:
    raise RuntimeError(
        "Usage: python delete_memory.py "
        "<memory-id> <exact-confirmation>"
    )

memory_id, confirmation = sys.argv[1:]
result = api_request(
    "DELETE",
    f"/memories/{quote(memory_id, safe='')}",
    body={"delete_confirmation": confirmation},
)

print(json.dumps(result, indent=2))

File: examples/javascript/delete-memory.mjs
import { memside } from "./memside-client.mjs";

const [memoryId, confirmation] = process.argv.slice(2);
if (!memoryId || !confirmation) {
  throw new Error(
    "Usage: node delete-memory.mjs <memory-id> <exact-confirmation>"
  );
}

const result = await memside(`/memories/${encodeURIComponent(memoryId)}`, {
  method: "DELETE",
  body: JSON.stringify({ delete_confirmation: confirmation }),
});

console.log(JSON.stringify(result, null, 2));
